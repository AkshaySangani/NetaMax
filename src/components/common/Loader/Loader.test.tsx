import { render, cleanup } from "@testing-library/react";
import { Loader } from "./index";
import { ILoaderProps } from "./index";

describe("Render Loader component", () => {
  let mock: ILoaderProps;

  beforeEach(() => {
    mock = {
      thickness: "4px",
      speed: "0.65s",
      emptyColor: "gray.200",
      color: "netaRed.500",
      size: "xl",
      delay: 0,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("Loader should be render correctly with default props", () => {
    const { container } = render(<Loader />);
    expect(container).toMatchSnapshot();
  });

  test("Loader should be render correctly with props", () => {
    const { container } = render(<Loader {...mock} />);
    expect(container).toMatchSnapshot();
  });

  test("Loader should render only if delay set to 0", () => {
    const delay = 1;
    const { container } = render(<Loader delay={delay} />);
    expect(container).toBeEmptyDOMElement();
  });
});
