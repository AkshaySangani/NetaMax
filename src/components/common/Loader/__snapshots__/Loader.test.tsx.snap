// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Loader component Loader should be render correctly with default props 1`] = `
<div>
  <div
    class="chakra-spinner css-166ypan"
  >
    <span
      class="css-f8n5zr"
    >
      Loading...
    </span>
  </div>
</div>
`;

exports[`Render Loader component Loader should be render correctly with props 1`] = `
<div>
  <div
    class="chakra-spinner css-166ypan"
  >
    <span
      class="css-f8n5zr"
    >
      Loading...
    </span>
  </div>
</div>
`;
